pkg/quota/admission/runonceduration/admission.go

```diff
@@ -0,0 +1,133 @@
+package runonceduration
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"reflect"
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	projectcache "github.com/openshift/origin/pkg/project/cache"
+	"github.com/openshift/origin/pkg/quota/admission/runonceduration/api"
+	"github.com/openshift/origin/pkg/quota/admission/runonceduration/api/validation"
+)
+
+func init() {
+	admission.RegisterPlugin("RunOnceDuration", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig, err := readConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		return NewRunOnceDuration(pluginConfig), nil
+	})
+}
+
+func readConfig(reader io.Reader) (*api.RunOnceDurationConfig, error) {
+	config := &api.RunOnceDurationConfig{}
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return config, nil
+	}
+	configBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	err = configlatest.ReadYAML(configBytes, config)
+	if err != nil {
+		return nil, err
+	}
+	errs := validation.ValidateRunOnceDurationConfig(config)
```

Comment:
They're used for different things. The Validate method is called after the plugin has been instantiated and after things like the project cache or openshift client have been set. It's meant to check whether the additional things you need besides what was given to you in initialization are present.

Here we are instantiating the plugin and we know enough to decide whether the configuration is valid or not and can fail early if not.